.github/workflows/code_quality.yml

```diff
@@ -0,0 +1,34 @@
+# This workflow will install Python dependencies, run tests and lint with a single version of Python
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+name: Code Quality
+
+on:
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+    name: Code Quality ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ "ubuntu-latest", "windows-latest" ]
```

Comment:
spend quite some time trying to resolve the issue. but failed. This needs a proper debugger ... on a linux/ubuntu machine for on windows this can't be reproduced.
I'm disabling the tests for now and created an issue for it: #25